hub/util/cache_chain.py

```diff
@@ -23,15 +25,48 @@ def get_cache_chain(provider_list: List[StorageProvider], size_list: List[int]):
         ProviderListEmptyError: If the provider list is empty.
         ProviderSizeListMismatch: If the len(size_list) + 1 != len(provider_list)
     """
-    if not provider_list:
+    if not storage_list:
         raise ProviderListEmptyError
-    if len(provider_list) <= 1:
-        return provider_list[0]
-    if len(size_list) + 1 != len(provider_list):
+    if len(storage_list) <= 1:
+        return storage_list[0]
+    if len(size_list) + 1 != len(storage_list):
         raise ProviderSizeListMismatch
-    provider_list.reverse()
+    storage_list.reverse()
```

Comment:
Can we remove these and use negative indexing instead? I think it's safer if we don't mutate the arguments